python/ray/util/spark/utils.py

```diff
@@ -357,11 +357,17 @@ def _get_num_physical_gpus():
             text=True,
             capture_output=True,
         )
-    except Exception as e:
-        raise RuntimeError(
+        return len(completed_proc.stdout.strip().split("\n"))
+    except Exception:
+        _logger.warning(
             "Running command `nvidia-smi` for inferring GPU devices list failed."
-        ) from e
-    return len(completed_proc.stdout.strip().split("\n"))
+        )
+        _logger.debug(
+            "'nvidia-smi --query-gpu=name --format=csv,noheader' command execution "
+            "failed.",
+            exc_info=True,
+        )
```

Comment:
why both debug and warning? if this can happen and is expected, we can use info, otherwise warning or error.